ffi/harfbuzz/src/lib.rs

```diff
@@ -69,606 +74,384 @@ use icu_properties::{GeneralCategory, Script};
 use icu_provider::prelude::*;
 use tinystr::tinystr;
 
-/// The total number of General Category values is fixed per
-/// https://www.unicode.org/policies/stability_policy.html :
-/// "The enumeration of General_Category property values is fixed. No new values will be added."
+use harfbuzz_traits::{
+    CombiningClassFunc, ComposeFunc, DecomposeFunc, GeneralCategoryFunc, MirroringFunc, ScriptFunc,
+};
+
+/// A single copyable UnicodeFuncs type that implements all of the `harfbuzz_trait` traits with compiled data.
 ///
-/// Despite General Category logically being a fixed enumeration, the Unicode
-/// Standard does not assign numeric identifiers to the possible General Category
-/// values. Both ICU4X and HarfBuzz do, but in different order. This table provides
-/// the permutation to go from the ICU4X number assignment in a Rust `enum` to
-/// the HarfBuzz number assignment in a C `enum`.
+/// Can be passed to the `harfbuzz` crate's `UnicodeFuncsBuilder`.
+#[cfg(feature = "compiled_data")]
+#[derive(Default, Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[allow(clippy::exhaustive_structs)] // unit struct
+pub struct UnicodeFuncs;
```

Comment:
Suggestion: since `UnicodeFuncs` is the base name of something in the harfbuzz crate, maybe name this differently?

- `AllUnicodeFuncs`
- `CompiledDataUnicodeFuncs`
- `CompiledUnicodeFuncs`
- `CompiledDataForHarfbuzz`
- ...